### PR TITLE
REF: de-duplicate ndarray[datetimelike] wrapping

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Review by jorisvandenbossche — COMMENTED** [review]
Personally, I find the "maybe" in the name useful here, as in my head, our "maybe" function maybe do something with the input, but otherwise return it not modified (I don't know if we consistently follow that logic though, for all our maybe_ functions). While "wrap_datetimelike" sounds as it always wraps a datetimelike (maybe_wrap_datetimelike?)

**Comment by jbrockmendel** [comment]
wrap_if_datetimelike?

File: pandas/core/construction.py [hunk 1]
```diff
@@ -402,6 +402,24 @@ def extract_array(obj: object, extract_numpy: bool = False) -> Union[Any, ArrayL
     return obj
 
 
+def wrap_if_datetimelike(arr):
```
Comment (jreback) [inline pandas/core/construction.py:405]: agree with joris that `maybe_*` is pretty much a convention and would go with that nomenclature.

Comment (jreback) [inline pandas/core/construction.py:405]: can you type the arg (and return dtype)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/construction.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/construction.py:405]: not really, its unrestricted

Comment (jbrockmendel) [reply pandas/core/construction.py:405]: maybe_do_x usually means "try to do x and fallback on failure".  i think wrap_if_datetimelike is more precise on this point.

Comment (jreback) [reply pandas/core/construction.py:405]: ok how about
ensure_datetimelike_index then

Comment (jreback) [reply pandas/core/construction.py:405]: aren't these *always* TDI/DTI on the return type? (sure input is just ArrayLike)

Comment (jbrockmendel) [reply pandas/core/construction.py:405]: no, we call it on arbitrary inputs in ops.array_ops

Comment (jreback) [reply pandas/core/construction.py:405]: ok then i am misunderstanding how this is used and its definetly maybe_wrap_if_datetimelike, its either ensure or maybe, really shouldn't be any other choices.

Comment (jbrockmendel) [reply pandas/core/construction.py:405]: ok will update

Comment (jbrockmendel) [reply pandas/core/construction.py]: renamed + greenish

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, type if possible in the future

**Pull request merged**